cuda_core/tests/conftest.py

```diff
@@ -21,22 +21,33 @@
 from cuda.core.experimental._utils import handle_return
 
 
+@pytest.fixture(scope="session", autouse=True)
+def always_init_cuda():
+    handle_return(driver.cuInit(0))
+
+
 @pytest.fixture(scope="function")
 def init_cuda():
+    # TODO: rename this to e.g. init_context
     device = Device()
     device.set_current()
     yield
     _device_unset_current()
 
 
 def _device_unset_current():
+    ctx = handle_return(driver.cuCtxGetCurrent())
+    if int(ctx) == 0:
+        # no active context, do nothing
+        return
```

Comment:
FYI, @ksimpson-work another issue caught by the CI (and also back in #261): A test could end early without a CUDA context set current, so we need to detect this at the test teardown time.